### PR TITLE
Add Flags for Energy Profilers - Issue #311

### PR DESCRIPTION
**Pull request by AkkiG2401**
Added flags for the energy profiler and structured the implementation to facilitate easy profiling of additional functions in the future. Fixes Issue #311.

### CODE REVIEW

File: baler/baler.py [hunk 1]
```diff
@@ -75,9 +75,87 @@ def main():
             + " not recognised. Use baler --help to see available modes."
         )
 
+def check_enabled_profilers(function, pytorch_profile, energy_profile, *args, **kwargs):
```
[the remark this answers is not in the document]
Comment (AkkiG2401) [reply baler/baler.py:78]: Hi @exook, I added this rough solution to allow running the profiling (if required in the future) on other functions as well. for example, if someone wants to run perform_compression they can just use check_enable_profilers(perform_compression, pytorch_profile, energy_profile, any other arguments).

**Comment by AkkiG2401** [comment]
> As stated in the issue I was hoping that this would be a flag in the config not in the run command.

I initially thought that it would be something run on demand by the experimenter, which is why I added it to the run command. However, I understand the convenience of having it in the config. If necessary, I can work on adding this option to the config for better flexibility 😊.

**Comment by exook** [comment]
It's a very interesting implementation, I'm afraid that the readability suffers, but I also agree that the config is becoming very bloated. Maybe @fritjof-b  can help us make a decision?

**Comment by AkkiG2401** [comment]
Hi @exook, as per feedback, I modified my initial approach and made the code a bit more readable. I had to change the profiler code to function wrappers (from decorators). This enabled a much cleaner implementation of the `check_enabled_profilers` function. I have also run Black to format the code properly.

**Comment by fritjof-b** [comment]
Do we need to have flags for energy profiling? When I've run profiling previously, I usually do it as a standalone program, while maybe adding target functions for the profiler. Such as:

```
$ MyProfilerSoftware SoftwareToTest <flags for actual program>
```

As an "outside" view, Baler's responsibility is to compress data--not profile the energy usage, this is just something that we _can_ do if we want.

Wdyt?

**Comment by exook** [comment]
Thanks for joining @fritjof-b, you have a very good point, this should probably have been standalone. As it is already in master I guess we should approve this request so that users can turn it off, and then at a later stage, we will clean up Baler and move things like this to standalone programs?

**Review by fritjof-b — APPROVED** [review]

**Pull request merged**